sycl/plugins/opencl/pi_opencl.hpp

```diff
@@ -115,4 +224,33 @@ inline const OpenCLVersion V3_0(3, 0);
 
 } // namespace OCLV
 
+// Define the types that are opaque in pi.h in a manner suitable for OpenCL
+// plugin
+
+struct _pi_device : _pi_object {
+  enum device_level {
+    ROOTDEVICE = 0,
+    SUBDEVICE = 1,
+    SUBSUBDEVICE = 2,
+    INVALID = -1
+  };
+  _pi_device(pi_platform Plt) : Platform{Plt} {
+    level = INVALID;
+    family = index = 0;
+    // NOTE: one must additionally call initialize() to complete
+    // PI device creation.
+  }
+  // PI platform to which this device belongs.
+  pi_platform Platform;
+
+  // Info stored for sub-sub device queue creation
+  device_level level;
+  pi_uint32 family; // SYCL queue family
+  pi_uint32 index;  // SYCL queue index inside a given family of queues
+
+  bool isRootDevice(void) { return level == ROOTDEVICE; }
+  bool isSubDevice(void) { return level == SUBDEVICE; }
```

Comment:
It's been more than a week since my review, so I've forgotten some amount of context, but looking into the conversation at https://github.com/intel/llvm/pull/7963#discussion_r1064906429 I'm realizing that I don't particularly like this interface.

`sub-sub-device->isSubDevice` is ambiguous in its meaning. It is a "sub-device" of a "sub-device", after all.

Reply:
Ah. I agree. I was not able to think of any clearer/easier way to represent this. May be a comment can help?

Reply:
changed isSubSubDevice to isCCS.